snuba/query/processors/uuid_column_processor.py

```diff
@@ -0,0 +1,150 @@
+        return Param(
+            "uuid_column" + suffix, ColumnMatch(None, None, self.__uuid_column_match)
+        )
+
+    def __init__(self, uuid_columns: List[str]) -> None:
+        self.__unique_uuid_columns = set(uuid_columns)
+        self.__uuid_column_match = Or(
+            [String(u_col) for u_col in self.__unique_uuid_columns]
+        )
+        self.uuid_in_condition = FunctionCallMatch(
+            None,
+            Or((String(ConditionFunctions.IN), String(ConditionFunctions.NOT_IN))),
+            (
+                Or((self.uuid_column_pattern(), self.formatted_uuid_pattern())),
+                Param("params", FunctionCallMatch(None, String("tuple"), None)),
+            ),
+        )
+        self.uuid_condition = FunctionCallMatch(
+            None,
+            Or([String(op) for op in FUNCTION_TO_OPERATOR]),
+            (
+                Or(
+                    (
+                        Param("literal_0", LiteralMatch(None, AnyOptionalString())),
+                        self.uuid_column_pattern("_0"),
+                        self.formatted_uuid_pattern("_0"),
+                    )
+                ),
+                Or(
+                    (
+                        Param("literal_1", LiteralMatch(None, AnyOptionalString())),
+                        self.uuid_column_pattern("_1"),
+                        self.formatted_uuid_pattern("_1"),
+                    )
+                ),
+            ),
+        )
+
+    def parse_uuid(self, lit: Expression) -> Expression:
+        if not isinstance(lit, Literal):
+            return lit
+
+        try:
+            parsed = uuid.UUID(lit.value)
+            return Literal(lit.alias, str(parsed))
+        except Exception:
+            return lit
+
+    def process_condition(self, exp: Expression) -> Expression:
+        result = self.uuid_in_condition.match(exp)
+        if result:
+            new_params = []
+            if result.optional_expression("formatted_uuid_column") is not None:
+                alias = result.string("format_alias")
+                column = result.expression("formatted_uuid_column")
+                new_params.append(Column(alias, column.table_name, column.column_name,))
+            else:
+                new_params.append(result.expression("uuid_column"))
+
+            params_fn = result.expression("params")
+            new_fn_params = []
+            for param in params_fn.parameters:
+                new_fn_params.append(self.parse_uuid(param))
+
+            new_params.append(
+                FunctionCall(
+                    params_fn.alias, params_fn.function_name, tuple(new_fn_params)
+                )
+            )
+
+            return binary_condition(exp.alias, exp.function_name, *new_params)
+
+        result = self.uuid_condition.match(exp)
+        print("UUID MATCH", result)
+        if result:
+            new_params = []
+            for suffix in ["_0", "_1"]:
```

Comment:
Do you need to distinguish between literal0 and literal1 to ensure the literal and the column are on the right side of the expression you build in case of non commutative operators ?
I wonder whether we need to support optimizing them considering we are talking about uuids only or whether we can keep this code a lot simpler by just focusing on equality checks.
If we focused on equality only we would not care which side of the expression the literal and the column are and we would not have to deal with two cases.